server/games/game.py

```diff
@@ -668,11 +668,14 @@ async def update_game_stats(self):
                 "WHERE lower(filename) = lower(%s)", (self.map_file_path, )
             )
             row = await result.fetchone()
+            is_generated = False
+            if self.map_file_path:
+                is_generated = "neroxis_map_generator" in self.map_file_path
 
             if row:
                 self.map_id = row['id']
 
-            if (not row or not row['ranked']) and self.validity is ValidityState.VALID:
+            if (not row or not row['ranked']) and self.validity is ValidityState.VALID and not is_generated:
```

Comment:
Lets make sure that it only checks `is_generated` if the map does not exist in the database. Otherwise it would be possible for the `ranked` column to be overridden by putting "neroxis_map_generator" in the name. Something like this:
```suggestion
            if self.validity is ValidityState.VALID and (row and not row.ranked or not row and not is_generated):
```

Will want to format it so the line is not too long.

Reply:
Changed